android/lib/src/main/java/com/swmansion/gesturehandler/TapGestureHandler.java

```diff
@@ -68,27 +88,74 @@ private void endTap() {
     }
   }
 
+  private static float getLastPointer(MotionEvent event, boolean isX) {
+    float offset = isX ? event.getRawX() - event.getX() : event.getRawY() - event.getY();
+    int excludeIndex = event.getActionMasked() == MotionEvent.ACTION_POINTER_UP ?
+            event.getActionIndex() : -1;
+      float sum = 0f;
+      int count = 0;
+      for (int i = 0, size = event.getPointerCount(); i < size; i++) {
+        if (i != excludeIndex) {
+          sum += (isX ? event.getX(i) : event.getY()) + offset;
+          count++;
+        }
+      }
+      return sum / count;
+  }
+
+  private boolean shouldFail() {
+    float dx = mLastX - mStartX + mOffsetX;
+    if (mMaxDeltaX != MAX_VALUE_IGNORE && Math.abs(dx) > mMaxDeltaX) {
+      return true;
+    }
+
+    float dy = mLastY - mStartY + mOffsetY;
+    if (mMaxDeltaY != MAX_VALUE_IGNORE && Math.abs(dy) > mMaxDeltaY) {
+      return true;
+    }
+
+    float dist = dy * dy + dx * dx;
+    return mMaxDist != MAX_VALUE_IGNORE && Math.abs(dist) > mMaxDist;
```

Comment:
`abs` is not needed here, also we should have `mMaxDistSq` (square of the distance). Please take a look on how its done in Pan handler (e.g. note that we keep `mMinDistSq` but have a setter for just minDist) and lets use the same patter here